File: my_jakarta/src/timer/AutomaticTimerBean.java
package timer;

import jakarta.annotation.Resource;
import jakarta.ejb.*;
import jakarta.enterprise.event.Event;
import jakarta.inject.Inject;
import java.util.Collection;

/**
 * @author Arun Gupta
 */
@Startup
@Singleton
public class AutomaticTimerBean {

    @Resource
    SessionContext ctx;

    @Inject
    Event<Ping> pingEvent;

    @Schedule(hour = "*", minute = "*", second = "*/5", info = "Every 5 second timer")
    public void printDate() {
        Collection<Timer> timers = ctx.getTimerService().getAllTimers();
        for (Timer t : timers) {
            pingEvent.fire(new Ping(t.getInfo().toString()));
        }
    }

}

File: my_jakarta_kt/src/timer/AutomaticTimerBean.kt
package timer

import jakarta.annotation.Resource
import jakarta.ejb.Schedule
import jakarta.ejb.SessionContext
import jakarta.ejb.Singleton
import jakarta.ejb.Startup
import jakarta.enterprise.event.Event
import jakarta.inject.Inject

/**
 * @author Arun Gupta
 */
@Startup
@Singleton
class AutomaticTimerBean {
    @Resource
    var ctx: SessionContext? = null

    @Inject
    var pingEvent: Event<Ping>? = null

    @Schedule(hour = "*", minute = "*", second = "*/5", info = "Every 5 second timer")
    fun printDate() {
        val timers = ctx!!.timerService.allTimers
        for (t in timers) {
            pingEvent!!.fire(Ping(t.info.toString()))
        }
    }
}